### PR TITLE
fix for osx compilation

### LINKED ISSUE #49: OSX build issues on conda
**Describe the bug**
Conda is unable to build the OSX version of the new version.

**Related issue**
(if exist)

**To Reproduce**
See build here: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=364581&view=logs&j=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&t=63fa899e-c160-5185-75c4-c5fd30c90374&l=257
```bash
  lib2/threading.c:113:5: warning: implicit declaration of function 'clock_gettime' is invalid in C99 [-Wimplicit-function-declaration]
      clock_gettime(CLOCK_REALTIME, &abstime);
      ^
  lib2/threading.c:113:19: error: use of undeclared identifier 'CLOCK_REALTIME'

```

**Expected behavior**
The build should succeed.

**Environment (please complete the following information):**
 - OS: macOS
 - Python 3.6-3.9
 - project version: 1.60.0

**Test data(please attach in the report):**
N.A.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by sarthakpati**
Hopefully will fix #49 but this needs to be tested since I don't have a mac machine.

### CODE REVIEW

File: lib2/threading.c [hunk 1]
```diff
@@ -32,6 +32,10 @@ int g_ppmd_threading_useless_symbol;
 #include <process.h>
 #include <errno.h>
 
+#ifdef __MACH__
```
Comment (cielavenir) [inline lib2/threading.c:35]: not inside `_WIN32`, please

```diff
diff --git a/lib2/threading.c b/lib2/threading.c
index 5e8eb73..a87e9cf 100644
--- a/lib2/threading.c
+++ b/lib2/threading.c
@@ -32,11 +32,6 @@ int g_ppmd_threading_useless_symbol;
 #include <process.h>
 #include <errno.h>
 
-#ifdef __MACH__
-#include <mach/clock.h>
-#include <mach/mach.h>
-#endif
-
 /* ===  Implementation  === */
 
 static unsigned __stdcall worker(void *arg) {
@@ -111,6 +106,11 @@ int PPMD_pthread_cond_timedwait(PPMD_pthread_cond_t *cond, PPMD_pthread_mutex_t
 #else
 /* ===   POSIX Systems   === */
 
+#ifdef __MACH__
+#include <mach/clock.h>
+#include <mach/mach.h>
+#endif
+
 int PPMD_pthread_cond_timedwait(PPMD_pthread_cond_t *cond, PPMD_pthread_mutex_t *mutex,
                                 unsigned long nsec) {
     struct timespec abstime;
```

**Comment by sarthakpati** [comment]
Does this look okay, @cielavenir?

**Comment by cielavenir** [comment]
@sarthakpati it builds now

**Comment by sarthakpati** [comment]
> @sarthakpati it builds now

That's great to hear! I am not sure why the 3.8 tests on Windows are failing, though [[ref](https://github.com/miurahr/pyppmd/pull/50/checks?check_run_id=3353461079)]

**Comment by cielavenir** [comment]
@sarthakpati the `8388608-1` is related to https://github.com/miurahr/pyppmd/issues/47, not sure `1048576-0` one though

**Comment by sarthakpati** [comment]
Oh okay. If you could merge this in and push an updated version to pip, I can resume the conda builds accordingly. 😄

**Comment by cielavenir** [comment]
@miurahr LGTM

**Comment by miurahr** [comment]
Thank you! I'd lie to squash and merge.

Note: Unfortunately Github actions virtual environment only support macOS 10.15 and macOS 11.

**Pull request merged**